src/client/Microsoft.Identity.Client/Internal/Requests/SilentRequest.cs

```diff
@@ -75,9 +89,10 @@ private async Task<IAccount> GetAccountFromParamsOrLoginHintAsync(AcquireTokenSi
         internal async override Task PreRunAsync()
         {
 
-            if (ServiceBundle.PlatformProxy.CanBrokerSupportSilentAuth() && 
-                AuthenticationRequestParameters.IsBrokerConfigured)
+            if (_canExecuteBroker)
```

Comment:
I think we need to look at naming. It's getting confusing as we have about 4 verbs around broker existence, invocation etc: canExecute, canSwitchTo, CanBrokerSupportSilentAuth and canInvoke. It's very difficult to reason on what the "code" does. The typical solution for this to define the concepts and associate a single name for each concept: 

1. broker is configured by the user 
2. broker can be invoked - if it exists & we can switch to it 
3. broker supports silent auth 
4. can execute broker which is a combination of 2 & 3

For 4, I would suggest `BrokerIsInstalledAndSupportsSilentAuth()` or smth similar. Have a look at the rest and see if you can consolidate naming.